common/src/BasePlugin.php

```diff
@@ -116,30 +116,59 @@ public function getProductMetaKeys(): array {
             WHERE
                 p.post_type = 'product'
                 AND pm.meta_key <> ''
+                AND pm.meta_value <> ''
         ");
         return array_map(
             function ($value) {
                 return [
                     'type' => 'meta_key',
-                    'name' => $value
+                    'name' => $value,
+                    'example_value' => substr($this->getMetaValue($value), 0, 15),
                 ];
             },
             $meta_keys
         );
     }
 
-    public function getProductKeys() {
+    public function getSelectOptions(string $selected_key, bool $suggested = false) {
+        $options = [];
+        foreach ($this->getProductKeys() as $key) {
+            if ($key['suggested'] != $suggested) {
+                continue;
+            }
+            $options[] = '<option value="' . $key['option_value'] . '" '
+                . ($key['option_value'] == $selected_key ? 'selected' : '') . '>'
+                . $key['label']
+                . '</option>';
+        }
+        return $options;
+    }
+
+    private function getProductKeys(): array {
         $custom_keys = array_merge($this->getProductMetaKeys(), $this->getCustomAttributes());
         return array_map(function ($value) {
             return [
                 'option_value' => $value['type'] . htmlentities($value['name']),
-                'label' => htmlentities($value['name']) . ' (' . $value['type'] . ')',
+                'label' => htmlentities($value['name']) . ' (e.g. "' . $value['example_value'] . '")',
+                'suggested' => $this->isValidGtin($value['example_value']),
             ];
         },
             $custom_keys
         );
     }
 
+    private function getMetaValue(string $meta_key) {
+        global $wpdb;
+        $sql = "
+            SELECT meta.meta_value
+            FROM {$wpdb->postmeta} meta
+            WHERE meta.meta_key = '{$meta_key}'
+            AND meta.meta_value <> ''
+            LIMIT 1;
```

Comment:
Maybe add `ORDER BY` here so you get the most recent value? `ORDER BY meta.meta_id DESC`

Reply:
Yes, this is good. I was thinking of somehow ordering first based on the meta_value that is similar to a GTIN standard, this would avoid if a wrong GTIN value is stored to be picked. 
But I was not sure how to do it and it also seemed overcomplicated, half of the plugin became gtin handling 😄

Reply:
Like this?  This will order values that match the regex first, and then other values.  But this is very inefficient because it has to order all the results (which might be hundreds of thousands) before returning the first one.

So you'd have to do it in two queries: first try to find one that looks like a number, and then fall back to one that doesn't.

Or, better, because it avoids scanning through all records if there are none with a number, select the first 10 and then use the first that is a number.  Or the first if there are no numbers.

But yeah, what you have now is fine to start with.  :)

~~~
ORDER BY
  meta.meta_value REGEXP "^[0-9]{13,}$" DESC,
  meta.meta_id
~~~